masonry/src/passes/accessibility.rs

```diff
@@ -22,19 +22,24 @@ fn build_accessibility_tree(
     rebuild_all: bool,
     scale_factor: f64,
 ) {
-    let _span = widget.item.make_trace_span().entered();
+    let _span = global_state
+        .trace
+        .access
+        .then(|| widget.item.make_trace_span().entered());
```

Comment:
I'm surprised just creating the spans has such a heavy performance impact. Are you sure we're not conflating them with something else?

Reply:
I believe that this is correct. Creating a filtered out span has very little performance impact, obviously, but the log file layer has no filtering.

A large amount of the work in using a span is done on creation - see https://github.com/tokio-rs/tracing/blob/bdbaf8007364ed2a766cca851d63de31b7c47e72/tracing-subscriber/src/registry/sharded.rs#L237-L269.

An alternative approach we could take would be to cache the span for each widget in its state. That approach would be likely to get most of the same advantages, but requires more fundamental changes. That is also done in https://github.com/bevyengine/bevy/pull/9390. I think I'll do that as a follow-up to this PR; I still think this is useful, because the current logs are extraordinarily spammy; but if we can maintain a proper tracing stack at all times, that would be neat.

Reply:
Yeah, I'm in favor of removing spammy logs, but removing spans (and/or making span code more verbose) bothers me a bit.

If we do caching, we could simply create the cached span during the WidgetAdded event. To avoid using an optional span we could create WidgetState with a default placeholder span.